contrib/kind.sh

```diff
@@ -34,6 +34,37 @@ delete()
   kind delete cluster --name ${KIND_CLUSTER_NAME:-ovn}
 }
 
+update_kubeconfig()
+{
+  # For IPv6 NO-HA Shared-Gateway Mode, it is a pure IPv6 network and no IPv4 connectivity.
+  # So need to updae the KubeConfig with an IPv6 address. In HA mode, there is a load-balancer
+  # in front which responds to the IPv4 requests.
+  if [ "$IP_FAMILY" == "ipv6" ] && [ "$OVN_GATEWAY_MODE" == "shared" ] && [ "$KIND_HA" == false ]; then
+    # Determine if API_URL has an IP Address or Container Name
+    SERVER=`echo "$API_URL" | sed -e "s@^https://@@" -e "s@:6443@@"`
+    if [[ "$SERVER" == *":"* ]]; then
+      # IP Address, so leave as is
+      SERVER=$API_URL
+    else
+      # Remove - Debugging
+      docker container inspect ${SERVER}
+
+      # Container Name, so lookup IP Address
+      GBLIPV6ADDR=`docker container inspect -f '{{ .NetworkSettings.Networks.kind.GlobalIPv6Address }}' ${SERVER}`
+      if [ "$GBLIPV6ADDR" != "<no value>" ]; then
+        SERVER="https://[$GBLIPV6ADDR]:6443"
+      else
+        GBLIPV6ADDR=`docker container inspect -f '{{ .NetworkSettings.Networks.bridge.GlobalIPv6Address }}' ${SERVER}`
+        if [ "$GBLIPV6ADDR" != "<no value>" ]; then
+          SERVER="https://[$GBLIPV6ADDR]:6443"
+        else
+          SERVER=$API_URL
+        fi
+      fi
+    fi
+    sed -i 's@\(server:\)\(.*\)@\1 '"${SERVER}"'@' ${KUBECONFIG}
```

Comment:
do we need to update `API_URL` for the other parts of kind setup that are using it?